packages/nimble-components/build/generate-icons/source/index.js

```diff
@@ -18,6 +19,18 @@ const trimSizeFromName = text => {
 const generatedFilePrefix = `// AUTO-GENERATED FILE - DO NOT EDIT DIRECTLY
 // See generation source in nimble-components/build/generate-icons\n`;
 
+const mdxFilePrefix = `import { Meta, Title, IconItem } from '@storybook/blocks';
+import * as iconStories from '../icon-base/tests/icons.stories';
+
+<Meta of={iconStories} />
+<Title of={iconStories} />
+`;
+
+const mdxIconTablePrefix = `
+| Icon | Synonyms |
+| ---- | -------- |
+`;
+
 const iconsDirectory = path.resolve(__dirname, '../../../src/icons');
```

Comment:
I put the `icons.mdx` file in the top-level icons directory, instead of a `tests` directory. Since the whole folder is autogenerated, maybe that's fine?

Reply:
The files generated should be treated as source code. I.e. could turn off the generation and check them in. So it's fine to be in icons but should go under icons/tests.

Reason behind the reason is files in any folder named tests/ get excluded from the final package automatically, so lean into that convention.

Reply:
Can see Malcolm's script for setting up the directory structure:
https://github.com/ni/nimble/blob/main/packages/nimble-blazor/build/generate-playwright-version-properties/source/index.js#L11